diff --git a/node_modules/gatsby-plugin-open-graph-images/index.js b/node_modules/gatsby-plugin-open-graph-images/index.js
index e9647d5..e7a4664 100644
--- a/node_modules/gatsby-plugin-open-graph-images/index.js
+++ b/node_modules/gatsby-plugin-open-graph-images/index.js
@@ -2,6 +2,7 @@ const { join } = require("path");
 const { config } = require("./src/config");
 
 exports.createOpenGraphImage = (createPage, options) => {
+  config.init(options)
   const { defaultSize, componentGenerationDir } = config.getConfig();
   const { path, component, context } = options;
 
diff --git a/node_modules/gatsby-plugin-open-graph-images/src/generator.js b/node_modules/gatsby-plugin-open-graph-images/src/generator.js
index 1ffd5ca..7895d6c 100644
--- a/node_modules/gatsby-plugin-open-graph-images/src/generator.js
+++ b/node_modules/gatsby-plugin-open-graph-images/src/generator.js
@@ -13,13 +13,12 @@ exports.generateOgImages = async (imageGenerationJobs) => {
     const { componentPath, imgPath, size } = imageGenerationJob;
     const componentUrl = `${servingUrl}/${componentPath}`;
 
-    await page.goto(componentUrl);
     await page.setViewport(size);
+    await page.goto(componentUrl, { 'waitUntil' : 'networkidle2' });
 
     ensureThatImageDirExists(imgPath);
     await page.screenshot({ path: imgPath, clip: { x: 0, y: 0, ...size } });
-
-    fs.unlinkSync(join("public", componentPath, "index.html"));
+    // fs.unlinkSync(join("public", componentPath, "index.html"));
 
     const printPath = `${imgPath.replace("public", "")} ${size.width}x${size.height}`;
     console.log(`🖼  created Image: ${printPath}`);
